csrc/ec_gen.cpp

```diff
@@ -183,16 +186,17 @@ JNIEXPORT void JNICALL Java_com_amazon_corretto_crypto_provider_EcGen_generateEc
         // Now that we have the params, extract the group from them (all we care about)
         CHECK_OPENSSL(group = EC_KEY_get0_group(ecParams));
 
-        // Build the structure which will hold our result
-        if(!likely(key = EC_KEY_new())) {
-            throw java_ex(EX_OOM, "Out of memory");
-        }
-
-        CHECK_OPENSSL(EC_KEY_set_group(key, group));
+        // Setup the temporary PKEY context with the extracted parameters.
+        CHECK_OPENSSL(tmpCtx = EVP_PKEY_CTX_new_id(EVP_PKEY_EC, NULL));
+        CHECK_OPENSSL(EVP_PKEY_paramgen_init(tmpCtx) == 1);
+        CHECK_OPENSSL(EVP_PKEY_CTX_set_ec_paramgen_curve_nid(tmpCtx, EC_GROUP_get_curve_name(group)) == 1);
+        CHECK_OPENSSL(EVP_PKEY_paramgen(tmpCtx, &pkey) == 1);
 
-        if (!likely(EC_KEY_generate_key(key))) {
-            throw_openssl("Unable to generate key");
```

Comment:
previously, on malloc failure and keygen failure, we would throw an exception up to the java code. what happens when a call to `CHECK_OPENSSL` fails? will that also result in an exception being thrown, or will the failure manifest in some other way?